haystack/schema.py

```diff
@@ -250,9 +250,14 @@ def from_json(cls, data: str, field_map: Optional[Dict[str, Any]] = None) -> Doc
         return cls.from_dict(dictionary, field_map=field_map)
 
     def __eq__(self, other):
+        content = getattr(other, "content", None)
+        if isinstance(content, pd.DataFrame):
+            is_content_equal = content.equals(self.content)
```

Comment:
An update to `Document.__eq__` is needed to work properly with tables stored as pandas dataframes.